wagtail/admin/tests/test_rich_text.py

```diff
@@ -491,3 +493,87 @@ def test_versioned_static_media(self):
         media_html = str(feature.media)
         self.assertRegex(media_html, r"feature.js\?v=(\w+)")
         self.assertRegex(media_html, r"feature.css\?v=(\w+)")
+
+
+class TestEntityFeatureChooserUrls(TestCase):
+    def test_chooser_urls_exist(self):
+        features = FeatureRegistry()
+        image = features.get_editor_plugin("draftail", "image")
+        embed = features.get_editor_plugin("draftail", "embed")
+        document = features.get_editor_plugin("draftail", "document-link")
+        link = features.get_editor_plugin("draftail", "link")
+
+        # imageChooser
+        self.assertIsNotNone(image.data.get("chooserUrls"))
+        self.assertEqual(
+            image.data["chooserUrls"]["imageChooser"],
+            reverse_lazy("wagtailimages_chooser:choose"),
+        )
+        # linkChoosers
+        self.assertIsNotNone(link.data.get("chooserUrls"))
+        self.assertEqual(
+            link.data["chooserUrls"]["pageChooser"],
+            reverse_lazy("wagtailadmin_choose_page"),
```

Comment:
We should probably use `reverse` within the unit tests, as we want to check the actual string. The unit tests should resolve this lazy object into a string but let's be sure we are getting the right content.

Remember we want to test the JSON, so maybe we should review this test to ensure we are testing the input's resolved JSON data.

Reply:
OK sorry, I think we do need this test as is (including reverse-lazy), however we ALSO need a new test that checks that the widget output contains the data with the resolved.

A place for this could be in `wagtail/admin/tests/test_rich_text.py`

See the test `TestDraftailWithFeatureOptions` we are checking for the data used for the feature in `test_settings_features_option_on_rich_text_field` -> `self.assertContains(response, '"type": "header-two"')`.

Around this line we can also check for the chooser URLs with something like. `"chooserUrls": {"imageChooser": "/admin/images/chooser/"}}` (need to find a way to ensure we are only checking for links ones, maybe a different test block that has the links feature enabled.

Hope this helps.